### PR TITLE
added trackComponent & tests 

### PR DESCRIPTION
**Pull request by adidoshiws**
### Ticket Link

---

### Related Links

---

### Description

- TrackComponent & test
- Updated itunesContainer index.js test
- updated snapshots
- play single track at a time

### Steps to Reproduce / Test

---

---

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's
containerPRDemo2

### Live Link

---

### CODE REVIEW

File: app/components/TrackCard/index.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  collectionName,
+  previewUrl,
+  trackId,
+  artworkUrl100,
+  dispatchCurrentTrack,
+  currentTrack
+}) {
+  const audioRef = useRef(null);
+  const [isPlaying, setIsPlaying] = useState(false);
+  // const [currentSong, setCurrentSong] = useState(null);
```
Comment (ritwik-wednesday) [inline app/components/TrackCard/index.js]: remove if not needed

File: app/components/TrackCard/index.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      if (currentTrack !== trackId) {
+        audioRef.current.pause();
+        setIsPlaying(false);
+      }
+    }
+  }, [currentTrack]);
+
+  const playHandler = () => {
+    // eslint-disable-next-line no-console
+    console.log(audioRef.current.paused);
```
Comment (ritwik-wednesday) [inline app/components/TrackCard/index.js]: remove logs

File: app/containers/ItunesContainer/reducer.js [hunk 3]
```diff
@@ -6,9 +6,10 @@ export const { Types: itunesContainerTypes, Creators: itunesContainerCreators }
   requestGetItuneTracks: ['trackName'],
   successGetItuneTracks: ['data'],
   failureGetItuneTracks: ['error'],
-  clearItuneTracks: {}
+  clearItuneTracks: {},
+  currentTrack: ['trackId']
```
Comment (ritwik-wednesday) [inline app/containers/ItunesContainer/reducer.js:10]: lets rename this to `setCurrentTrack` as this is an action, easier to understand

File: app/components/TrackCard/index.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    if (audioRef.current.paused) {
+      dispatchCurrentTrack(trackId);
+      audioRef.current.play();
+      setIsPlaying(true);
+    } else {
+      dispatchCurrentTrack(null);
+      audioRef.current.pause();
+      setIsPlaying(false);
+    }
+    setIsPlaying(!isPlaying);
```
Comment (ritwik-wednesday) [inline app/components/TrackCard/index.js]: we are already setting `isPlaying` in the if/else block, wouldn't this just revert the state change and set it to previous state

File: app/components/TrackCard/index.js (hunk 1 above)
Comment (adidoshiws) [reply app/components/TrackCard/index.js]: removed!

File: app/components/TrackCard/index.js (hunk 2 above)
Comment (adidoshiws) [reply app/components/TrackCard/index.js]: yes done!

File: app/components/TrackCard/index.js (hunk 4 above)
Comment (adidoshiws) [reply app/components/TrackCard/index.js]: Correct, I was doing that to toggle btn text, but that's not needed.

File: app/containers/ItunesContainer/reducer.js (hunk 3 above)
Comment (adidoshiws) [reply app/containers/ItunesContainer/reducer.js:10]: sure, that would be a better naming.

File: app/components/T/index.js [hunk 5]
```diff
@@ -15,6 +15,7 @@ const StyledText = styled.p`
   && {
     ${(props) => props.marginBottom && `margin-bottom: ${props.marginBottom}px;`};
     ${(props) => props.font()};
+    width: 260px;
```
Comment (ritwik-wednesday) [inline app/components/T/index.js]: use `rem` units in css instead of `px`

File: app/components/TrackCard/index.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+import styled from 'styled-components';
+import { Col, Button } from 'antd';
+import T from '@components/T';
+import If from '@components/If';
+import { isEmpty } from 'lodash';
+import { Link } from 'react-router-dom';
+
+const CustomCard = styled(Col)`
+  && {
+    margin: 20px 0;
```
Comment (ritwik-wednesday) [inline app/components/TrackCard/index.js]: use rem

File: app/containers/ItunesContainer/reducer.js (hunk 3 above)
Comment (ritwik-wednesday) [reply app/containers/ItunesContainer/reducer.js:10]: Please make the change

File: app/components/TrackCard/index.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+          <T data-testid="collectionName" id="track_collection_name" values={{ collectionName: collectionName }} />
+        </If>
+      </Link>
+      <If
+        condition={!isEmpty(previewUrl)}
+        otherwise={<T data-testid="preview-unavailable" id="track_preview_unavailable" />}
+      >
+        <audio ref={audioRef} data-testid="preview" id="track_preview" src={previewUrl}></audio>
+        <Button type="primary" onClick={() => playHandler(trackId)}>
+          {isPlaying ? 'Pause' : 'Preview'}
```
Comment (ritwik-wednesday) [inline app/components/TrackCard/index.js]: lets use translations for text across the app

File: app/containers/ItunesContainer/index.js [hunk 8]
```diff
@@ -40,56 +88,116 @@ export function ItunesContainer({ dispatchItuneTracks, dispatchClearItuneTracks,
...
+            <div>
+              <T id="search_query_text" values={{ trackName }} />
+            </div>
+          </If>
+          <If condition={resultCount !== 0}>
+            <div>
+              <T id="matching_tracks" values={{ resultCount }} />
+            </div>
+          </If>
+          <Grid>
```
Comment (ritwik-wednesday) [inline app/containers/ItunesContainer/index.js]: we can remove this. We already set `Grid` as parentComponent in `For` component

File: app/components/T/index.js (hunk 5 above)
Comment (adidoshiws) [reply app/components/T/index.js]: changed to rem

File: app/components/TrackCard/index.js (hunk 6 above)
Comment (adidoshiws) [reply app/components/TrackCard/index.js]: done

File: app/components/TrackCard/index.js (hunk 7 above)
Comment (adidoshiws) [reply app/components/TrackCard/index.js]: yes, used translations for the button text

File: app/containers/ItunesContainer/index.js (hunk 8 above)
Comment (adidoshiws) [reply app/containers/ItunesContainer/index.js]: removed it

**Review by ritwik-wednesday — APPROVED** [review]

**Pull request merged**